assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -318,6 +318,56 @@ const ExternalActivity: React.FC<any> = (props) => {
     }
   };
 
+  const getInterestedVars = (vars: any) => {
+    const interested = Object.keys(vars).filter((ms: any) => {
+      const isMine = ms.indexOf(`stage.${id}.`) === 0;
+      if (!isMine) {
+        return false;
+      }
+      const internalValue = externalActivityMap.get(ms);
+      let mineValue = vars[ms];
+      let internalVal: any = internalValue?.value;
+
+      if (ms.type === CapiVariableTypes.BOOLEAN && typeof internalVal === 'string') {
+        mineValue = JSON.stringify(mineValue);
+      }
+      if (ms.type === CapiVariableTypes.NUMBER && typeof internalVal === 'string') {
+        mineValue = JSON.stringify(mineValue);
+      }
+      if (
+        typeof vars[ms] === 'object' &&
+        Array.isArray(vars[ms]) &&
+        typeof internalVal === 'string'
+      ) {
+        internalVal = new CapiVariable({
+          key: ms.key,
+          type: ms.type,
+          value: internalVal,
+        });
+        if (Array.isArray(internalVal.value) && Array.isArray(mineValue)) {
+          return JSON.stringify(internalVal.value) !== JSON.stringify(mineValue);
+        }
+      }
+      if (
+        typeof vars[ms] === 'object' &&
+        Array.isArray(vars[ms]) &&
+        typeof internalVal === 'object' &&
+        Array.isArray(internalVal)
+      ) {
+        return JSON.stringify(internalVal) !== JSON.stringify(mineValue);
+      }
+      if (mineValue == '' && internalVal == null) return false;
+      return !internalValue || internalVal != mineValue;
+    });
+    if (interested.length) {
```

Comment:
you don't need this branch here, the reduce of empty length array is going to return `{}` also